components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/device/DeviceEndpoint.java

```diff
@@ -0,0 +1,118 @@
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+@Path("/device_authorize")
+public class DeviceEndpoint {
+
+    private static final Log log = LogFactory.getLog(DeviceEndpoint.class);
+
+    @POST
+    @Path("/")
+    @Consumes("application/x-www-form-urlencoded")
+    @Produces("application/json")
+    public Response authorize(@Context HttpServletRequest request, @Context HttpServletResponse response)
+            throws IdentityOAuth2Exception {
+
+        String clientId = request.getParameter(Constants.CLIENT_ID);
+        JSONObject errorResponse = new JSONObject();
+        if (StringUtils.isBlank(clientId)) {
+            errorResponse.put(Constants.ERROR, DeviceErrorCodes.INVALID_REQUEST)
+                    .put(Constants.ERROR_DESCRIPTION,"Request missing required parameters");
+            Response.ResponseBuilder respBuilder = Response.status(HttpServletResponse.SC_BAD_REQUEST);
+            return respBuilder.entity(errorResponse.toString()).build();
+        } else {
+            if (validateClientId(clientId)) {
+                String userCode = GenerateKeys.getKey(Constants.KEY_LENGTH);
+                String deviceCode = UUID.randomUUID().toString();
+                String scope = request.getParameter(Constants.SCOPE);
+                String redirectionUri = IdentityUtil.getServerURL("/authenticationendpoint/device.do", false, false);
+                String redirectionUriComplete = redirectionUri + "?user_code=" + userCode;
+                DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO().insertDeviceFlowParameters(deviceCode,
+                        userCode, clientId, Constants.EXPIRES_IN_VALUE, Constants.INTERVAL_VALUE);
+                DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO().storeDeviceFlowScopes(scope, deviceCode);
+
+                JSONObject jsonObject = new JSONObject();
+                jsonObject.put(Constants.DEVICE_CODE, deviceCode)
+                        .put(Constants.USER_CODE, userCode)
+                        .put(Constants.VERIFICATION_URI, redirectionUri)
+                        .put(Constants.VERIFICATION_URI_COMPLETE, redirectionUriComplete)
+                        .put(Constants.EXPIRES_IN, stringValueInSeconds(Constants.EXPIRES_IN_VALUE))
+                        .put(Constants.INTERVAL, stringValueInSeconds(Constants.INTERVAL_VALUE));
+                Response.ResponseBuilder respBuilder = Response.status(HttpServletResponse.SC_OK);
+
+                return respBuilder.entity(jsonObject.toString()).build();
+
+            } else {
+                errorResponse.put(Constants.ERROR, DeviceErrorCodes.UNAUTHORIZED_CLIENT)
+                        .put(Constants.ERROR_DESCRIPTION,"No registered client with the client id.");
+                Response.ResponseBuilder respBuilder = Response.status(HttpServletResponse.SC_UNAUTHORIZED);
+                return respBuilder.entity(errorResponse.toString()).build();
+            }
+        }
```

Comment:
Shall we use guard if  clause here.
Ex:
if (StringUtils.isBlank(clientId)){
return somevalue;
}
if(!validateClientId(clientId)){
return somevalue;
}

return somevalue;